src/main/java/tira/Storage.java

```diff
@@ -0,0 +1,152 @@
+        FileReader fileReader = null;
+
+        try {
+            if (!file.exists()) { //create new file if the file doesn't exist
+                file.getParentFile().mkdirs();
+                file.createNewFile();
+                return tasks;
+            } else {// a file exists
+                fileReader = new FileReader(file);
+                BufferedReader bufferedReader = new BufferedReader(fileReader);
+                ArrayList<String> lines = new ArrayList<>();
+                String line = bufferedReader.readLine();
+                while (line != null) {
+                    lines.add(line);
+                    if (bufferedReader.readLine() == null) {
+                        break;
+                    }
+                }
+                return this.convertToTaskList(lines);
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("File not Found");
+        } catch (IOException a) {
+            System.out.println("Error while file loading");
+        }
+        for (Task i : tasks) {
+            System.out.println(i);
+        }
+        return tasks;
+    }
+
+    /*
+    Takes the task in tasks, converts to String, then writes to the file
+     */
+    public void save(ArrayList<Task> tasks) throws IOException {
+        FileWriter writer = new FileWriter(filePath);
+        for (int i = 0; i < tasks.size(); i++) {
+            Task currentTask = tasks.get(i);
+            String string = this.convertTaskToString(tasks.get(i));
+            writer.write(string);
+        }
+        writer.close();
+    }
+
+    //@@hansneddyanto-reused
+    // Hans is my friend and he has allowed me to refer to parts of his code.
+    public String convertTaskToString(Task task) {
+        String output = "";
+        if (task instanceof ToDo) {
+            output += "T | ";
+            if (task.getIsDone()) {
+                output += "1";
+            } else {
+                output += "0";
+            }
+            output += " | " + task.getDescription();
+        }
+        if (task instanceof Deadline) {
+            output += "D |";
+            Deadline deadline = (Deadline) task;
+            if (deadline.getIsDone()) {
+                output += "1";
+            } else {
+                output += "0";
+            }
+            output += " | " + deadline.getDescription() + " | " + deadline.getEndDate();
+
+        }
+        if (task instanceof Event) {
+            output += "D |";
+            Event event = (Event) task;
+            if (event.getIsDone()) {
+                output += "1 ";
+            } else {
+                output += "0 ";
+            }
+            output +=  " | " + event.getDescription() + " | " + event.getStartDate() +
+                    " | " + event.getEndDate();
+        }
+        return output;
+    }
+
+    public ArrayList<Task> convertToTaskList(ArrayList<String> taskStringList) {
```

Comment:
Maybe a javadoc comment here would be nice?

Reply:
Agreed! I feel it would provide much clarity explaining what the purpose of the method, as well as what the input `taskStringList` represents.